sys/include/net/ng_ipv6/addr.h

```diff
@@ -225,38 +234,107 @@ static inline bool ng_ipv6_addr_is_unspecified(const ng_ipv6_addr_t *addr)
  *
- * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
- *          RFC 4291, section 2.7
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.3">
+ *          RFC 4291, section 2.5.3
  *      </a>
  *
  * @param[in] addr  An IPv6 address.
  *
- * @return  true, if @p addr is multicast address,
+ * @return  true, if @p addr is loopback address,
  * @return  false, otherwise.
  */
-static inline bool ng_ipv6_addr_is_multicast(const ng_ipv6_addr_t *addr)
+static inline bool ng_ipv6_addr_is_loopback(const ng_ipv6_addr_t *addr)
 {
-    return (addr->u8[0] == 0xff);
+    return (addr->u64[0].u64 == 0) &&
+           (byteorder_ntohll(addr->u64[1]) == 1);
 }
 
 /**
- * @brief   Checks if @p addr is a loopback address.
+ * @brief   Check if @p addr is global unicast address.
  *
- * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.3">
- *          RFC 4291, section 2.5.3
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.4">
+ *          RFC 4291, section 2.5.4
  *      </a>
  *
  * @param[in] addr  An IPv6 address.
  *
- * @return  true, if @p addr is loopback address,
+ * @return  true, if @p addr is global unicast address,
  * @return  false, otherwise.
  */
-static inline bool ng_ipv6_addr_is_loopback(const ng_ipv6_addr_t *addr)
+bool ng_ipv6_addr_is_global_unicast(const ng_ipv6_addr_t *addr);
+
+/**
+ * @brief   Checks if @p addr is a IPv4-compatible IPv6 address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.5.1">
+ *          RFC 4291, section 2.5.5.1
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is an IPv4-compatible IPv6 address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_ipv4_compat(const ng_ipv6_addr_t *addr)
 {
     return (addr->u64[0].u64 == 0) &&
-           (byteorder_ntohll(addr->u64[1]) == 1);
+           (addr->u32[2].u32 == 0);
+}
+
+/**
+ * @brief   Checks if @p addr is a IPv4-mapped IPv6 address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.5.1">
+ *          RFC 4291, section 2.5.5.2
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is an IPv4-compatible IPv6 address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_ipv4_mapped(const ng_ipv6_addr_t *addr)
+{
+    return ((addr->u64[0].u64 == 0) &&
+            (addr->u16[4].u16 == 0) &&
+            (byteorder_ntohs(addr->u16[5]) == 0xFFFF));
+}
+
+/**
+ * @brief   Checks if @p addr is a site-local unicast address.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.5.7">
+ *          RFC 4291, section 2.5.7
+ *      </a>
+ *
+ * @param[in] addr  An IPv6 address.
+ *
+ * @return  true, if @p addr is a site-local unicast address,
+ * @return  false, otherwise.
+ */
+static inline bool ng_ipv6_addr_is_site_local(const ng_ipv6_addr_t *addr)
+{
+    return ((byteorder_ntohs(addr->u16[0]) & NG_IPV6_ADDR_SITE_LOCAL_PREFIX) ==
```

Comment:
`ffc0` would fall through the cracks, too. Also, since the link-local equivalent is checking the link-local multicast prefix too, this one should do it too.